### PR TITLE
vendor mpb@v7.1.4

### PR DESCRIPTION
**Pull request by vrothberg**
Fixes a race condition leading to a deadlock.

Thanks to @mtrmac and @vbauerster for fixing the issue!

Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

**Review by Luap99 — COMMENTED** [review]
LGTM

**Comment by vrothberg** [comment]
@containers/podman-maintainers PTAL

File: test/e2e/search_test.go [hunk 1]
```diff
@@ -477,6 +477,6 @@ registries = ['{{.Host}}:{{.Port}}']`
 		search := podmanTest.Podman([]string{"search", "--limit", "130", "registry.redhat.io/rhel"})
 		search.WaitWithDefaultTimeout()
 		Expect(search).Should(Exit(0))
-		Expect(len(search.OutputToStringArray())).To(Equal(131))
+		Expect(len(search.OutputToStringArray())).To(BeNumerically(">", 100))
```
Comment (Luap99) [inline test/e2e/search_test.go:480]: This should check that this is not over 131 since the limit is set to 130.

Comment (vrothberg) [reply test/e2e/search_test.go]: Much better :

**Pull request merged**